File: Practica8/incremento.py
class A:
    def __init__(self, x, z):
        self.x = x
        self.z = z

    def incrementaXZ(self):
        self.x += 1
        self.z += 1

    def incrementaZ(self):
        self.z += 1

class B:
    def __init__(self, y, z):
        self.y = y
        self.z = z

    def incrementaYZ(self):
        self.y += 1
        self.z += 1

    def incrementaZ(self):
        self.z += 1

class D(A, B):
    def __init__(self, x, y, z):
        A.__init__(self, x, z)
        B.__init__(self, y, z)

    def incrementaXYZ(self):
        self.x += 1
        self.y += 1
        self.z += 1

if __name__ == "__main__":
    obj = D(5, 7, 0)
    print(f"Antes: x={obj.x}, y={obj.y}, z={obj.z}")
    obj.incrementaXYZ()
    print(f"Después de incrementaXYZ: x={obj.x}, y={obj.y}, z={obj.z}")

File: Practica8/incremento.java
class A {
    public int x;
    public int z;

    public A(int x, int z) {
        this.x = x;
        this.z = z;
    }

    public void incrementaXZ() {
        x++;
        z++;
    }

    public void incrementaZ() {
        z++;
    }
}

class B {
    public int y;
    public int z;

    public B(int y, int z) {
        this.y = y;
        this.z = z;
    }

    public void incrementaYZ() {
        y++;
        z++;
    }

    public void incrementaZ() {
        z++;
    }
}

public class D {
    public int x, y, z;

    public D(int x, int y, int z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    // Métodos equivalentes a los de A y B
    public void incrementaXZ() {
        x++;
        z++;
    }

    public void incrementaYZ() {
        y++;
        z++;
    }

    public void incrementaZ() {
        z++;
    }

    public void incrementaXYZ() {
        x++;
        y++;
        z++;
    }

    public void mostrar(String mensaje) {
        System.out.println(mensaje + ": x=" + x + ", y=" + y + ", z=" + z);
    }

    public static void main(String[] args) {
        D obj = new D(1, 2, 3);
        obj.mostrar("Antes");

        obj.incrementaXYZ();
        obj.mostrar("Después de incrementaXYZ");

        obj.incrementaXZ();
        obj.mostrar("Después de incrementaXZ");

        obj.incrementaYZ();
        obj.mostrar("Después de incrementaYZ");

        obj.incrementaZ();
        obj.mostrar("Después de incrementaZ");
    }
}
